### PR TITLE
test: Check that peers with forcerelay permission are not asked to feefilter

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

File: test/functional/p2p_feefilter.py [hunk 1]
```diff
@@ -9,12 +9,16 @@
 
 from test_framework.messages import MSG_TX, msg_feefilter
 from test_framework.mininode import mininode_lock, P2PInterface
-from test_framework.test_framework import BitcoinTestFramework
+from test_framework.test_framework import (
+    BitcoinTestFramework,
+    assert_equal,
+)
```
Comment (jonatack) [inline test/functional/p2p_feefilter.py:15]: This runs, but ISTM no other tests import this way and instead do:
```python
from test_framework.test_framework import BitcoinTestFramework
from test_framework.util import assert_equal
```

**Review by jonatack — COMMENTED** [review]
ACK fa473a0448817004a6271fbee926eb4a2a4258a4 modulo unusual test import

File: test/functional/p2p_feefilter.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/p2p_feefilter.py]: Whoopsie. Fixed!

**Comment by jonatack** [comment]
ACK faccf0a3b6f1

**Review by vahidjalaliii — APPROVED** [review]

**Comment by vahidjalaliii** [comment]
#19198

**Comment by MarcoFalke** [comment]
Force pushed a move-only to avoid a merge conflict (no rebase done). Should be trivial to re-ACK

**Review by jonatack — COMMENTED** [review]
Code review re-ACK faaac4e move-only change of two class member functions in test_framework.py since my last review per `git range-diff 4b5c919 faccf0a faaac4e`, idem since rebase per `git diff fa39586 faaac4e`. Edit: verified p2p_feefilter and p2p_permissions functional tests run green locally.

**Comment by MarcoFalke** [comment]
Sorry, I had to rebase due to a silent merge conflict, now that `mininode_lock` is no longer a recursive lock.

**Comment by jonatack** [comment]
re-ACK fac63eb move-only change of two class member functions in test_framework.py and rebases since my review @ faccf0a per `git range-diff 4b5c919 faccf0a fac63eb`. Verified p2p_feefilter and p2p_permissions functional tests are running :green_circle: locally.

**Pull request merged**